plugins/src/main/java/com/google/firebase/gradle/plugins/report/UnitTestReport.kt

```diff
@@ -0,0 +1,288 @@
+    var name =
+      job
+        .get("name")
+        .getAsString()
+        .split("\\(:".toRegex())
+        .dropLastWhile { it.isEmpty() }
+        .toTypedArray()[1]
+    name = name.substring(0, name.length - 1) // Remove trailing ")"
+    var status = TestReport.Status.OTHER
+    if (job.get("status").asString == "completed") {
+      if (job.get("conclusion").asString == "success") {
+        status = TestReport.Status.SUCCESS
+      } else {
+        status = TestReport.Status.FAILURE
+      }
+    }
+    val url = job.get("html_url").getAsString()
+    return TestReport(name, type, status, commit, url)
+  }
+
+  private fun request(path: String): JsonObject {
+    return request(path, JsonObject::class.java)
+  }
+
+  private fun <T> request(path: String, clazz: Class<T>): T {
+    return request(URI.create(URL_PREFIX + path), clazz)
+  }
+
+  /**
+   * Abstracts away paginated calling. Naively joins pages together by merging root level arrays.
+   */
+  private fun <T> request(uri: URI, clazz: Class<T>): T {
+    val request =
+      HttpRequest.newBuilder()
+        .GET()
+        .uri(uri)
+        .header("Authorization", "Bearer $apiToken")
+        .header("X-GitHub-Api-Version", "2022-11-28")
+        .build()
+    try {
+      val response = client.send(request, HttpResponse.BodyHandlers.ofString())
+      val body = response.body()
+      if (response.statusCode() >= 300) {
+        System.err.println(response)
+        System.err.println(body)
+      }
+      val json: T = GSON.fromJson(body, clazz)
+      if (json is JsonObject) {
+        // Retrieve and merge objects from other pages, if present
+        response.headers().firstValue("Link").ifPresent { link: String ->
+          val parts = link.split(",".toRegex()).dropLastWhile { it.isEmpty() }
+          for (part in parts) {
+            if (part.endsWith("rel=\"next\"")) {
+              // <foo>; rel="next" -> foo
+              val url =
+                part
+                  .split(">;".toRegex())
+                  .dropLastWhile { it.isEmpty() }
+                  .toTypedArray()[0]
+                  .split("<".toRegex())
+                  .dropLastWhile { it.isEmpty() }
+                  .toTypedArray()[1]
+              val p = request<JsonObject>(URI.create(url), JsonObject::class.java)
+              for (key in json.keySet()) {
+                if (json.get(key).isJsonArray && p.has(key) && p.get(key).isJsonArray) {
+                  json.getAsJsonArray(key).addAll(p.getAsJsonArray(key))
+                }
+              }
+              break
+            }
+          }
+        }
+      }
+      return json
+    } catch (e: IOException) {
+      throw RuntimeException(e)
+    } catch (e: InterruptedException) {
+      throw RuntimeException(e)
+    }
+  }
+
+  companion object {
+    private val PR_NUMBER_MATCHER: Pattern = Pattern.compile(".*\\(#([0-9]+)\\)")
```

Comment:
Drive by comment with my personal rule: every regex string should have a comment above it explaining it what it does :)